mono/mini/liveness.c

```diff
@@ -66,12 +69,12 @@ mono_ptrset_add (MonoPtrSet *set, gpointer val)
 		GHashTable *tmp = g_hash_table_new (NULL, NULL);
 		for (int i = 0; i < INLINE_SIZE; ++i)
 			g_hash_table_insert (tmp, set->data [i], set->data [i]);
-		set->data [0] = tmp;
```

Comment:
This line debatable actually, was ok before, as can assign to void*  from non-void*, just not the other way around. It might follow the rules better in terms of reading the written union member though or people might prefer it.